Fix for #785: channel with self

pcppcp opened the pull request:
This fixes a situation where someone attempts to create channel with both peers set to the same address.  Instead of "Duplicate channel" more useful exception info is raised. 
The requirement is also checked in the solidity contract.

LefterisJP commented on raiden/smart_contracts/ChannelManagerContract.sol:
```diff
@@ -82,6 +82,7 @@ contract ChannelManagerContract {
     /// @param settle_timeout The settle timeout in blocks
     /// @return The address of the newly created NettingChannelContract.
     function newChannel(address partner, uint settle_timeout) returns (address) {
+        require(msg.sender != partner);
```
This is not required. It's checked in the `NettingChannelContract` constructor.

https://github.com/raiden-network/raiden/blob/e17d96db375d31b134ae7b4e2ad2c1f905b47857/raiden/smart_contracts/NettingChannelContract.sol#L27

LefterisJP commented on raiden/smart_contracts/ChannelManagerLibrary.sol:
```diff
@@ -33,6 +33,8 @@ library ChannelManagerLibrary {
     function newChannel(Data storage self, address partner, uint settle_timeout)
         returns (address)
     {
+        require(msg.sender != partner);
```
This is not required. It's checked in the `NettingChannelContract` constructor .. and you also put it in the channel library contract above which calls this code.

https://github.com/raiden-network/raiden/blob/e17d96db375d31b134ae7b4e2ad2c1f905b47857/raiden/smart_contracts/NettingChannelContract.sol#L27

LefterisJP commented on raiden/tests/integration/test_channel_to_self.py:
```diff
@@ -0,0 +1,31 @@
+@pytest.mark.parametrize('number_of_nodes', [1])
+@pytest.mark.parametrize('channels_per_node', [0])
+@pytest.mark.parametrize('blockchain_type', ['geth'])
+def test_channel_with_self(raiden_network, settle_timeout):
```
Why make a new test file for this? Please add it to one of the other test files.

pcppcp replied:
OK, I was wondering if I should put it elsewhere..

LefterisJP commented:
@pcppcp I would also suggest you don't throw a `ValueError` but create a specific exception and check for it [here](https://github.com/raiden-network/raiden/blob/91263b3ec608c8be0f038fb9bcb202649c4eb3da/raiden/api/rest.py#L246) so that when someone does this mistake from an API call they get a proper error and not a `500` Internal Server error.

pcppcp commented:
Changed as proposed in review:

- removed solidity check for peer adresses
- REST api returns 409 instead of 500 on same address for channel peers
- included test in another test batch

LefterisJP commented:
@pcppcp Looks fine to me now. But I would update the "peer" wording to "participant" to be consistent with the rest of the documentation/codebase.
